ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+        /// satisfies some criteria. For example, this class is used to report model inputs
+        /// and outputs, for searching operations and tensors by name, for setting shape etc.
+        ///
+        /// Place can refer to Tensor, Input Edge, Input Port, Operation, Output Port, Output Edge
+        ///
+        ///                [Tensor A]
+        ///                    |
+        ///                    | [Input Edge]
+        ///                    |
+        ///                    V
+        ///           -------------------
+        ///           [  [Input Port 0] ]
+        ///           [                 ]
+        ///           [   Operation A   ]
+        ///           [                 ]
+        ///           [ [Output Port 0] ]
+        ///           -------------------
+        ///                    |
+        ///                    | [Output Edge]
+        ///                    |
+        ///                    V
+        ///                [Tensor B]
+        ///                    |
+        ///                    | [Input Edge]
+        ///                    |
+        ///                    V
+        ///           -------------------
+        ///           [  [Input Port 0] ]
+        ///           [                 ]
+        ///           [   Operation B   ]
+        ///           [                 ]
+        ///           [ [Output Port 0] ]
+        ///           -------------------
+        ///                    |
+        ///                    | [Output Edge]
+        ///                    |
+        ///                    V
+        ///                [Tensor C]
+        ///
+        class FRONTEND_API Place
+        {
+        public:
+            typedef std::shared_ptr<Place> Ptr;
+
+            virtual ~Place() = default;
+
+            /// \brief All associated names (synonyms) that identify this place in the graph in a
+            /// framework specific way
+            ///
+            /// \return A vector of strings each representing a name that identifies this place in
+            /// the graph. Can be empty if there are no names associated with this place or name
+            /// cannot be attached.
+            virtual std::vector<std::string> get_names() const;
+
+            /// \brief Returns references to all operation nodes that consume data from this place
+            /// \note It can be called for any kind of graph place searching for the first consuming
+            /// operations.
+            ///
+            /// \param outputPortIndex If place is an operational node it specifies which output
+            /// port should be considered. It is optional if place has only one output port
+            ///
+            /// \return A vector with all operation node references that consumes data from this
+            /// place
+            virtual std::vector<Ptr> get_consuming_operations(int outputPortIndex = -1) const;
```

Comment:
Output ports are only exists inside operation but Place looks so generic so I would expect that `get_consuming_operations` should return all consumers from all output ports if Place represents Node in a graph. And to get consumers for specific output port I think Place should represent specific port but not a Node.

Reply:
@slyalin Can you please comment on this?

Reply:
To get consumers, a given place should represent some unique data in a graph, not multiple output ports. It can be a node with a single output port, or output port (place itself can represent output port, in this case outputPortIndex shouldn't be used -- and actually it was my ask to separate this method to two: without a parameter that should work for any places, and with parameter that should work for operations only), or a tensor (please refer to the picture inside the comment at the beginning of this file).

Reply:
@slyalin So, what is the final decision? Should I split these method into 2 ones?

Reply:
Yes.